crates/autopilot/src/solvable_orders.rs

```diff
@@ -235,10 +235,14 @@ impl SolvableOrdersCache {
             latest_settlement_block: db_solvable_orders.latest_settlement_block,
             orders: orders
                 .into_iter()
-                .map(|order| {
-                    let quote = db_solvable_orders.quotes.get(&order.metadata.uid.into());
-                    let protocol_fees = self.protocol_fee.get(&order, quote);
-                    boundary::order::to_domain(order, protocol_fees)
+                .filter_map(|order| {
+                    if let Some(quote) = db_solvable_orders.quotes.get(&order.metadata.uid.into()) {
+                        let protocol_fees = self.protocol_fee.get(&order, quote);
+                        Some(boundary::order::to_domain(order, protocol_fees))
```

Comment:
I think it would read slightly nicer if `ProtocolFee` would return the domain order type.

Reply:
Fixed